test/SignAndDelete.test.js

```diff
@@ -0,0 +1,308 @@
+      expectedOAuthNonceSize = chance.string();
+
+      getConfig.mockImplementation(() => {
+        return {
+          firstLegUri: expectedFirstLegUri,
+          thirdLegUri: expectedThirdLegUri,
+          clientKey: expectedClientKey,
+          clientSecret: expectedClientSecret,
+          oAuthVersion: expectedOAuthVersion,
+          authorizeCallbackUri: expectedAuthorizeCallbackUri,
+          oAuthSignatureMethod: expectedOAuthSignatureMethod,
+          oAuthNonceSize: expectedOAuthNonceSize,
+        };
+      });
+
+      await underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("creates an OAuth with same values from config", () => {
+      expect(OAuth.OAuth).toBeCalledWith(
+        expectedFirstLegUri,
+        expectedThirdLegUri,
+        expectedClientKey,
+        expectedClientSecret,
+        expectedOAuthVersion,
+        expectedAuthorizeCallbackUri,
+        expectedOAuthSignatureMethod,
+        expectedOAuthNonceSize,
+        expect.anything()
+      );
+    });
+  });
+
+  describe("is called", () => {
+    beforeEach(async () => {
+      await underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("passed JD Accept header", () => {
+      expect(OAuth.OAuth).toBeCalledWith(
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        { Accept: "application/vnd.deere.axiom.v3+json" }
+      );
+    });
+  });
+
+  describe("optionalAuthorizationCallbackUri is provided", () => {
+    let expectedAuthorizeCallbackUri;
+
+    beforeEach(async () => {
+      expectedAuthorizeCallbackUri = chance.string();
+
+      await underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string(),
+        expectedAuthorizeCallbackUri
+      );
+    });
+
+    it("creates an OAuth with the provided optionalAuthorizationCallbackUri", () => {
+      expect(OAuth.OAuth).toBeCalledWith(
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        expectedAuthorizeCallbackUri,
+        undefined,
+        undefined,
+        expect.anything()
+      );
+    });
+  });
+
+  describe("OAuth delete returns an error", () => {
+    let expectedError;
+    let result;
+
+    beforeEach(() => {
+      expectedError = chance.string();
+      jest.spyOn(OAuth, "OAuth").mockImplementation(() => ({
+        delete: (link, accessToken, accessTokenSecret, callback) => {
+          callback(expectedError, undefined, { statusCode: 200 });
+        },
+      }));
+
+      result = underTest.doSignAndDelete(
+        chance.string(),
+        chance.string(),
+        chance.string()
+      );
+    });
+
+    it("throws an error when there is an error in the OAuth delete response", async () => {
```

Comment:
```suggestion
    it("should reject with the same error", async () => {
```
Don't need the when since it's defined in the beforeEach. It's not technically throwing here just rejecting.